iroh-net/src/discovery/pkarr.rs

```diff
@@ -174,6 +181,48 @@ impl PublisherService {
     }
 }
 
+/// Resolve node info using a pkarr relay.
+#[derive(derive_more::Debug, Clone)]
+pub struct PkarrResolver {
+    pkarr_client: PkarrRelayClient,
+}
+
+impl PkarrResolver {
+    /// Create a new config with a pkarr relay URL.
+    pub fn new(pkarr_relay: Url) -> Self {
+        Self {
+            pkarr_client: PkarrRelayClient::new(pkarr_relay),
+        }
+    }
+
+    /// Create a config that resolves using the n0 dns server through [`N0_DNS_PKARR_RELAY`].
+    pub fn n0_dns() -> Self {
+        let pkarr_relay: Url = N0_DNS_PKARR_RELAY.parse().expect("url is valid");
+        Self::new(pkarr_relay)
+    }
+}
+
+impl Discovery for PkarrResolver {
+    fn resolve(
+        &self,
+        _ep: Endpoint,
+        node_id: NodeId,
+    ) -> Option<BoxStream<'static, Result<DiscoveryItem>>> {
```

Comment:
why I had to add 'static here explicitly, I have no idea:
```
cargo check
    Blocking waiting for file lock on build directory
    Checking iroh-net v0.18.0 (/home/dvc/iroh/iroh-net)
error: `impl` item signature doesn't match `trait` item signature
   --> iroh-net/src/discovery/pkarr.rs:206:5
    |
206 |       fn resolve(&self, _ep: Endpoint, node_id: NodeId) -> Option<BoxStream<Result<DiscoveryItem>>> {
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ found `fn(&'1 PkarrResolver, endpoint::Endpoint, iroh_base::key::PublicKey) -> std::option::Option<Pin<Box<(dyn futures_lite::Stream<Item = Result<DiscoveryItem, anyhow::Error>> + std::marker::Send + '1)>>>`
    |
   ::: iroh-net/src/discovery.rs:43:5
    |
43  | /     fn resolve(
44  | |         &self,
45  | |         _endpoint: Endpoint,
46  | |         _node_id: NodeId,
47  | |     ) -> Option<BoxStream<Result<DiscoveryItem>>> {
    | |_________________________________________________- expected `fn(&'1 PkarrResolver, endpoint::Endpoint, iroh_base::key::PublicKey) -> std::option::Option<Pin<Box<(dyn futures_lite::Stream<Item = Result<DiscoveryItem, anyhow::Error>> + std::marker::Send + 'static)>>>`
    |
    = note: expected signature `fn(&'1 PkarrResolver, endpoint::Endpoint, iroh_base::key::PublicKey) -> std::option::Option<Pin<Box<(dyn futures_lite::Stream<Item = Result<DiscoveryItem, anyhow::Error>> + std::marker::Send + 'static)>>>`
               found signature `fn(&'1 PkarrResolver, endpoint::Endpoint, iroh_base::key::PublicKey) -> std::option::Option<Pin<Box<(dyn futures_lite::Stream<Item = Result<DiscoveryItem, anyhow::Error>> + std::marker::Send + '1)>>>`
    = help: the lifetime requirements from the `impl` do not correspond to the requirements in the `trait`
    = help: verify the lifetime relationships in the `trait` and `impl` between the `self` argument, the other inputs and its output

error: could not compile `iroh-net` (lib) due to 1 previous error
```
```
rustc --version
rustc 1.78.0 (9b00956e5 2024-04-29)
```

Reply:
🤷